libs/server/Resp/AdminCommands.cs

```diff
@@ -17,42 +17,19 @@ namespace Garnet.server
     /// </summary>
     internal sealed unsafe partial class RespServerSession : ServerSessionBase
     {
-        private (RespCommand, byte) ParseAdminCommands()
-        {
-            byte* ptr = recvBufferPtr + readHead;
-            byte* cmdPtr = null;
-            int csize = 0;
-            if (!RespReadUtils.ReadPtrWithLengthHeader(ref cmdPtr, ref csize, ref ptr, recvBufferPtr + bytesRead))
-                return (RespCommand.NONE, 0);
-
-            var command = new ReadOnlySpan<byte>(cmdPtr, csize);
-            var cmd = RespCommand.NONE;
-            byte subcmd = 0;
-
-            if (command.SequenceEqual(CmdStrings.ECHO)) return (RespCommand.ECHO, 0);
-            else if (command.SequenceEqual(CmdStrings.REPLICAOF)) return (RespCommand.REPLICAOF, 0);
-            else if (command.SequenceEqual(CmdStrings.SECONDARYOF)) return (RespCommand.SECONDARYOF, 0);
-            else if (command.SequenceEqual(CmdStrings.CONFIG)) return (RespCommand.CONFIG, 0);
-            else if (command.SequenceEqual(CmdStrings.CLIENT)) return (RespCommand.CLIENT, 0);
-            else if (command.SequenceEqual(CmdStrings.MEMORY)) return (RespCommand.MEMORY, 0);
-            else if (command.SequenceEqual(CmdStrings.MONITOR)) return (RespCommand.MONITOR, 0);
-
-            return (cmd, subcmd);
-        }
-
-        private bool ProcessAdminCommands<TGarnetApi>(ReadOnlySpan<byte> command, ReadOnlySpan<byte> bufSpan, int count, ref TGarnetApi storageApi)
+        private bool ProcessAdminCommands<TGarnetApi>(RespCommand command, ReadOnlySpan<byte> bufSpan, int count, ref TGarnetApi storageApi)
```

Comment:
nit: Can we rename "count" to something more explanatory (like "tokenCount")?

Reply:
I would say we keep that for the next iteration of parsing refactor. The argument handling for command executors probably has to change completely.